url8db16c7d62_Arsaell_EdT_tgz-pJ8-DATA_MainJ8/src/GUI/WeekPanel.java

```diff
@@ -0,0 +1,141 @@
+				{new Integer(2), null, null, null, null, null, null, null},
+				{new Integer(3), null, null, null, null, null, null, null},
+				{new Integer(4), null, null, null, null, null, null, null},
+				{new Integer(5), null, null, null, null, null, null, null},
+				{new Integer(6), null, null, null, null, null, null, null},
+				{new Integer(7), null, null, null, null, null, null, null},
+				{new Integer(8), null, null, null, null, null, null, null},
+				{new Integer(9), null, null, null, null, null, null, null},
+				{new Integer(10), null, null, null, null, null, null, null},
+				{new Integer(11), null, null, null, null, null, null, null},
+				{new Integer(12), null, null, null, null, null, null, null},
+				{new Integer(13), null, null, null, null, null, null, null},
+				{new Integer(14), null, null, null, null, null, null, null},
+				{new Integer(15), null, null, null, null, null, null, null},
+				{new Integer(16), null, null, null, null, null, null, null},
+				{new Integer(17), null, null, null, null, null, null, null},
+				{new Integer(18), null, null, null, null, null, null, null},
+				{new Integer(19), null, null, null, null, null, null, null},
+				{new Integer(20), null, null, null, null, null, null, null},
+				{new Integer(21), null, null, null, null, null, null, null},
+				{new Integer(22), null, null, null, null, null, null, null},
+				{new Integer(23), null, null, null, null, null, null, null},
+			},
+			columnNames
+		) {
+			Class[] columnTypes = new Class[] {
+				Integer.class, Boolean.class, Boolean.class, Boolean.class, Boolean.class, Boolean.class, Boolean.class, Boolean.class
+			};
+			public Class getColumnClass(int columnIndex) {
+				return columnTypes[columnIndex];
+			}
+		});
+		
+		for (int i = 0 ; i < table.getColumnModel().getColumnCount() ; i++)	{
+
+			table.getColumnModel().getColumn(i).setResizable(false);
+			table.getColumnModel().getColumn(i).setMinWidth(65);
+			table.getColumnModel().getColumn(i).setMaxWidth(100);
+		}
+		setLayout(new BorderLayout(0, 0));
+		
+		JLabel lblNewLabel = new JLabel("Veuillez lister les heures de cours de la semaine.");
+		lblNewLabel.setHorizontalAlignment(SwingConstants.CENTER);
+		add(lblNewLabel, BorderLayout.NORTH);
+		
+		
+		table.setColumnSelectionAllowed(true);
+		table.setCellSelectionEnabled(true);
+
+		JScrollPane scrollPane = new JScrollPane(table);
+		add(scrollPane, BorderLayout.CENTER);
+		
+		JButton btnValider = new JButton("Valider");
+		btnValider.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				//System.out.println("WeekPanel.valider() : " + container.ds);
+				container.ds.setDefaultWeek(updateWeekTable());
+				
+				container.allowTimeableTabs(true);
+				//System.out.println("WeekPanel.valider() : " + container.ds);
+			}
+		});
+		btnValider.setVerticalAlignment(SwingConstants.TOP);
+		add(btnValider, BorderLayout.SOUTH);
+		
+	}
+
+	private WeekTable updateWeekTable()	{
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.